### DISCUSSION TITLE
when tiffsave,how to set label image on image information?

### DISCUSSION THREAD

**Discussion by junqiangchen** [libvips/libvips#4050] [Q&A]
Hi,
i have convert many jpg files to pyramid tiff,but i want to add label image to tiff information label property.
![微信图片_20240716122727](https://github.com/user-attachments/assets/303d96f4-9516-4375-80f3-1db94be4f223)
here is my code:
`def generate_tif(image_dir, save_dir):
    ###########
    # image_dir: 扫描后小图存放的路径根目录
    # save_dir: 保存拼接后图像的路径
    ###########

    if not os.path.exists(save_dir):
        os.makedirs(save_dir)

    try:
        dir_name = os.path.basename(image_dir)
        print(f'Is generating {dir_name}.tif')
        if os.path.exists(os.path.join(save_dir, f'{dir_name}.tif')):
            raise Exception(f'已经被拼接过的文件:{dir_name}.tif')

        dir_name = os.path.basename(image_dir)
        print(f'Is generating {dir_name}.tif')
        patch_dir = os.path.join(image_dir, 'tile/focus0/0')
        if not os.path.exists(patch_dir):
            patch_dir = os.path.join(image_dir, 'tile/focus1/0')
            if not os.path.exists(patch_dir):
                raise Exception('没有找到合适的拼接图层')

        img_list = os.listdir(patch_dir)

        x_list = np.unique(np.array([int(img_name[:3]) for img_name in img_list]))
        y_list = np.unique(np.array([int(img_name[4:7]) for img_name in img_list]))
        x_max = x_list.max() + 1
        y_max = y_list.max() + 1
        patch_width = 1024
        patch_height = 1024

        vimgs = []
        for x in range(x_max):
            for y in range(y_max):
                img_path = os.path.join(patch_dir, '{:03d}x{:03d}.jpg'.format(x, y))
                if os.path.exists(img_path):
                    img = pyvips.Image.new_from_file(img_path)
                else:
                    img = np.zeros((patch_width, patch_width, 3), dtype=np.uint8)
                    img.fill(255)
                    img = pyvips.Image.new_from_array(img)
                vimgs.append(img)

        vimg = pyvips.Image.arrayjoin(vimgs, across=y_max)

        # 此处请把PhysicalSizeX,Y 修改为扫描的实际mpp
        vimg.set_type(pyvips.GValue.gstr_type, 'image-description',
                      f"""<?xml version="1.0" encoding="UTF-8"?>
                    <OME xmlns="http://www.openmicroscopy.org/Schemas/OME/2016-06"
                        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                        xsi:schemaLocation="http://www.openmicroscopy.org/Schemas/OME/2016-06 http://www.openmicroscopy.org/Schemas/OME/2016-06/ome.xsd">
                        <Image ID="Image:0">
                            
                            <Pixels DimensionOrder="XYCZT"
                                    PhysicalSizeX=0.25
                                    PhysicalSizeY=0.25
                                    PhysicalSizeXUnit=μm
                                    PhysicalSizeXUnit=μm
                            </Pixels>
                        </Image>
                    </OME>"""
                      )

        # 同样修改xrex与yres，0.25为真实的mpp
        vimg.tiffsave(os.path.join(save_dir, f'{dir_name}.tif'), tile=True, tile_width=1024, tile_height=1024,
                      pyramid=True,
                      compression='jpeg',
                      xres=1000.0 / 0.25,
                      yres=1000.0 / 0.25,
                      properties=True,
                      Q=80)
    except Exception:
        import traceback
        print(traceback.print_exc())`

can anyone help me?

thank you!!

**Comment by jcupitt** [comment]
Hello @junqiangchen, libtiff does not have a label property. Perhaps you mean SVS, not TIFF?

SVS is a non-standard TIFF variant used by Aperio slide scanners. You need to write them yourself, it's not easy, perhaps a week's work for a skilled developer. See for example:

https://stackoverflow.com/questions/75365808/is-it-possible-to-create-svs-files-at-all
